org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/batch/Main.java

```diff
@@ -4753,20 +4694,12 @@ public void performCompilation() {
 		String setting = System.getProperty("jdt.compiler.useSingleThread"); //$NON-NLS-1$
 		this.batchCompiler.useSingleThread = setting != null && setting.equals("true"); //$NON-NLS-1$
 
-		if (this.compilerOptions.complianceLevel >= ClassFileConstants.JDK1_6
-				&& this.compilerOptions.processAnnotations) {
-			if (checkVMVersion(ClassFileConstants.JDK1_6)) {
-				initializeAnnotationProcessorManager();
-				if (this.classNames != null) {
-					this.batchCompiler.setBinaryTypes(processClassNames(this.batchCompiler.lookupEnvironment));
-				}
-			} else {
-				// report a warning
-				this.logger.logIncorrectVMVersionForAnnotationProcessing();
-			}
-			if (checkVMVersion(ClassFileConstants.JDK9)) {
-				initRootModules(this.batchCompiler.lookupEnvironment, environment);
+		if (this.compilerOptions.processAnnotations) {
+			initializeAnnotationProcessorManager();
+			if (this.classNames != null) {
+				this.batchCompiler.setBinaryTypes(processClassNames(this.batchCompiler.lookupEnvironment));
 			}
+			initRootModules(this.batchCompiler.lookupEnvironment, environment);
```

Comment:
> have you checked if `initRootModules()` works (= does nothing) at 1.8?

The code was always executed before, because `checkVMVersion(ClassFileConstants.JDK9)` checked the current VM is >= 9 , quote:
> `Return true if and only if the running VM supports the given minimal version `

... and that is the case for a very long time :-)
Also `initRootModules()` looks like it only does something if `rootModules` and `limitedModules` are non empty, which is why it works on 1.8.

Reply:
Scratching my head on this till I noticed this discussion. Sounds good. Thanks